RIOT-OS/RIOT
# riotboot_serial: enter bootloader mode by pin

### Contribution description

The `riotboot_serial` bootloader did not have the feature yet that it remains in bootloader mode when a certain pin is pulled / a button is pressed.

This makes flashing much easier as we don't have to rely on the right timing anymore. 


### Testing procedure

On a board that has a button, flash `bootloaders/riotboot_serial`.

Now reset the board and press the bootloader button (BTN0). After a bit, flash any other application with `PROGRAMMER=riotboot_serial RESET=`. 

The programmer should be able to connect to the board. Without this, it would attempt to load the first image after a timeout (and fail). 


### Issues/PRs references
